tests/backup/backup_test.go

```diff
@@ -922,14 +922,14 @@ var _ = Describe("{ShareBackupWithUsersAndGroups}", func() {
 		Step("Add users to group", func() {
 			log.InfoD("Adding users to groups")
 			var wg sync.WaitGroup
-			for i := 0; i < len(users); i++ {
+			for i, user := range users {
 				groupIndex := i / groupSize
 				wg.Add(1)
-				go func(i, groupIndex int) {
-					err := backup.AddGroupToUser(users[i], groups[groupIndex])
+				go func(userName string, groupIndex int) {
+					defer wg.Done()
+					err := backup.AddGroupToUser(userName, groups[groupIndex])
```

Comment:
I dont see any difference in the logic from previous one.
Earlier index is being used to get username from users now directly getting using range.
is there anything Im missing?

Reply:
Before, the i variable is shared between all the goroutines, which could lead to race conditions if not properly synchronized. The use of a range expression now avoids this problem by automatically creating a new variable for each iteration of the loop, which is then passed to the goroutine.

Reply:
No. You are right. There is no change in logic. The difference in behaviour was seen because of the environment issue. Anyway, this feels a better way to represent, so I will keep it as it is.